index.js

```diff
@@ -24,67 +26,81 @@ function getRandomNumber(min, max) {
 
 function checkGuess() {
   // Get value from guess input element
-  const guess = parseInt(guessInput.value, 10);
+  let guess = parseInt(guessInput.value, 10);
   attempts = attempts + 1;
 
+  if (guess < 1) {
+    // Display a message for guesses lower than 0
+    numberLessThanZero.style.display = "";
+    return;
+  }
+
+  if (guess > 99) {
+    // Display a message for guesses higher than 99
+    numberHigherThan99.style.display = "";
+    return; // Exit the function to prevent further execution
+  }
+
   hideAllMessages();
 
   if (guess === targetNumber) {
-    numberOfGuessesMessage.style.display = '';
+    numberOfGuessesMessage.style.display = "";
     numberOfGuessesMessage.innerHTML = `You made ${attempts} guesses`;
 
-    correctMessage.style.display = '';
+    correctMessage.style.display = "";
 
     submitButton.disabled = true;
     guessInput.disabled = true;
-  }
-
-  if (guess !== targetNumber) {
+  } else {
     if (guess < targetNumber) {
-      tooLowMessage.style.display = '';
+      tooLowMessage.style.display = ""; //Fixed the display message
     } else {
-      tooLowMessage.style.display = '';
+      tooHighMessage.style.display = ""; //Fixed the display message
     }
 
     const remainingAttempts = maxNumberOfAttempts - attempts;
 
-    numberOfGuessesMessage.style.display = '';
-    numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} guesses remaining`;
+    numberOfGuessesMessage.style.display = "";
+    numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} 
+    ${remainingAttempts === 1 ? "guess" : "guesses"} remaining`;
```

Comment:
This is great solution for the reach goal, but template strings can become hard to read if you cram too much logic in them. Another option would be something like:
```suggestion
    const guessString = remainingAttempts === 1 ? "guess" : "guesses";
    numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} ${guessString} remaining`;
```